python3_tools/get_repos.py

```diff
@@ -158,6 +158,29 @@ def is_oep7_compliant(repo):
     # Check to see if openedx.yaml has an oeps section
     return is_oep_compliant(repo, "oep-7")
 
+@cache_to_disk(1)
+def is_oep18_compliant(repo):
+    """
+    OEP-18 is the one about Python Dependency Management
+        -mainly includes implementing make upgrade and placing all the dependencies in requirement directory
+    returns (bool,str): whether it's compliant and reason it's not if bool is false.
+    """
+    return is_oep_compliant(repo, "oep-18")
+
+@cache_to_disk(1)
+def might_be_oep18_compliant(repo):
+    """
+    Use alternate indicators to see if a repo is oep18 complaint. Checks to see if requirement folder exists or not
+    
+    returns (bool,str): whether it might compliant and reason we think so if it's true.
+    """
+    try:
+        head_directory_content = repo.get_contents("requirements")
```

Comment:
Rather than checking if the requirements folder exists.  What about checking if the 'upgrade' target existis in a makefile?  In a lot of cases the folder will exist even if we are not oep-18 compliant.  But usually the 'upgrade' target won't.